packages/infrastructure/src/collection/saleor/infrastructure/get-collections-ids-by-slugs-infra.ts

```diff
@@ -0,0 +1,73 @@
+import { err, ok } from "@nimara/domain/objects/Result";
+
+import {
+  type GetCollectionsIDsBySlugsInfra,
+  type SaleorCollectionServiceConfig,
+} from "#root/collection/types";
+import { graphqlClient } from "#root/graphql/client";
+
+import { CollectionsIDsBySlugsDocument } from "../graphql/queries/generated";
+
+export const getCollectionsIDsBySlugsInfra =
+  ({
+    apiURI,
+    logger,
+  }: SaleorCollectionServiceConfig): GetCollectionsIDsBySlugsInfra =>
+  async ({ channel, slugs, options }) => {
+    try {
+      logger.debug("Fetching the collections ID's from Saleor", {
+        slugs,
+        channel,
+      });
+
+      if (!slugs || slugs.length === 0) {
+        return ok({ results: null });
+      }
+
+      const result = await graphqlClient(apiURI).execute(
+        CollectionsIDsBySlugsDocument,
+        {
+          variables: {
+            channel,
+            slugs,
+          },
+          operationName: "CollectionsIDsBySlugsQuery",
+          options,
+        },
+      );
+
+      if (!result.ok) {
+        logger.error("Failed to fetch collections ID's from Saleor", {
+          error: result.errors,
+        });
```

Comment:
Add more context to the error message:
```suggestion
        logger.error("Failed to fetch collections ID's from Saleor", {
          error: result.errors,
          channel,
          slugs
        });
```